watchtower-plugin/src/retrier.rs

```diff
@@ -16,174 +16,242 @@ use crate::wt_client::WTClient;
-                            if !pending_appointments.is_empty() {
-                                // If there are pending appointments by the time we remove the retrier we send them back through the channel
-                                // so they are not missed. Notice this is unlikely given the map is checked before exiting `retry_tower`, but it
-                                // can happen.
-                                log::info!(
-                                    "Some data was missed while retrying {}. Adding it back",
-                                    tower_id
-                                );
-                                for locator in retrier.pending_appointments.lock().unwrap().drain()
-                                {
-                                    state.unreachable_towers.send((tower_id, locator)).unwrap();
-                                }
-                            } else {
-                                log::info!("Retry strategy succeeded for {}", tower_id);
-                                state.set_tower_status(tower_id, crate::TowerStatus::Reachable);
-                            }
-                        }
-                        Err(e) => {
-                            log::warn!("Retry strategy gave up for {}. {}", tower_id, e);
-                            // Notice we'll end up here after a permanent error. That is, either after finishing the backoff strategy
-                            // unsuccessfully or by manually raising such an error (like when facing a tower misbehavior)
-                            if let Some(tower) = state.towers.get_mut(&tower_id) {
-                                if tower.status.is_temporary_unreachable() {
-                                    log::warn!("Setting {} as unreachable", tower_id);
-                                    state.set_tower_status(
-                                        tower_id,
-                                        crate::TowerStatus::Unreachable,
-                                    );
-                                }
-                            } else {
-                                log::info!("Skipping retrying abandoned tower {}", tower_id);
-                            }
+                Err(TryRecvError::Empty) => {
+                    // Keep only running retriers and retriers ready to be started/re-started.
+                    // This will remove failed ones and ones finished successfully and have no pending appointments.
+                    //
+                    // Note that a failed retrier could have received some new appointments to retry. In this case, we don't try to send
+                    // them because we know that that tower is unreachable. We most likely received these new appointments because the tower
+                    // was marked as temporary unreachable during the retrial process so the sender didn't know that the tower is unreachable.
```

Comment:
I'd reword this to: "We most likely received these new appointments while the tower was still flagged as temporarily unreachable when cleaning up after giving up retrying."